tests/test_phoenix.rs

```diff
@@ -3440,3 +3561,298 @@ async fn test_phoenix_basic_with_raw_base_unit_adjustment() {
+        self_trade_behavior: SelfTradeBehavior::Abort,
+        match_limit: None,
+        client_order_id: 0,
+        use_only_deposited_funds: false,
+        last_valid_slot: None,
+        last_valid_unix_timestamp_in_seconds: None,
+        fail_silently_on_insufficient_funds: true,
+    };
+    let new_order_ix =
+        create_new_order_instruction(market, &maker.user.pubkey(), base_mint, quote_mint, &params);
+
+    sdk.client
+        .sign_send_instructions(vec![new_order_ix], vec![&maker.user])
+        .await
+        .unwrap();
+
+    println!("Placing ask order (using only deposited funds) for 1 SOL");
+
+    let deposit_ix = create_deposit_funds_instruction(
+        market,
+        &maker.user.pubkey(),
+        &meta.base_mint,
+        &meta.quote_mint,
+        &DepositParams {
+            quote_lots_to_deposit: 0,
+            base_lots_to_deposit: meta.raw_base_units_to_base_lots_rounded_down(1.0),
+        },
+    );
+
+    let params = OrderPacket::Limit {
+        side: Side::Ask,
+        price_in_ticks: Ticks::new(meta.float_price_to_ticks_rounded_down(10.0)),
+        num_base_lots: BaseLots::new(meta.raw_base_units_to_base_lots_rounded_down(1_f64)),
+        self_trade_behavior: SelfTradeBehavior::Abort,
+        match_limit: None,
+        client_order_id: 0,
+        use_only_deposited_funds: true,
+        last_valid_slot: None,
+        last_valid_unix_timestamp_in_seconds: None,
+        fail_silently_on_insufficient_funds: true,
+    };
+    let new_order_ix =
+        create_new_order_with_free_funds_instruction(market, &maker.user.pubkey(), &params);
+
+    sdk.client
+        .sign_send_instructions(vec![deposit_ix, new_order_ix], vec![&maker.user])
+        .await
+        .unwrap();
+
+    let market_start = sdk.get_market_orderbook(market).await.unwrap();
+    assert_eq!(market_start.asks.len(), 3);
+
+    // This order should fail silently
+    let params = OrderPacket::Limit {
+        side: Side::Ask,
+        price_in_ticks: Ticks::new(meta.float_price_to_ticks_rounded_down(10.0)),
+        num_base_lots: BaseLots::new(meta.raw_base_units_to_base_lots_rounded_down(2_f64)),
+        self_trade_behavior: SelfTradeBehavior::Abort,
+        match_limit: None,
+        client_order_id: 0,
+        use_only_deposited_funds: false,
+        last_valid_slot: None,
+        last_valid_unix_timestamp_in_seconds: None,
+        fail_silently_on_insufficient_funds: true,
+    };
+
+    let new_order_ix =
+        create_new_order_instruction(market, &maker.user.pubkey(), base_mint, quote_mint, &params);
+
+    sdk.client
+        .sign_send_instructions(vec![new_order_ix], vec![&maker.user])
+        .await
+        .unwrap();
+
+    let market_end = sdk.get_market_orderbook(market).await.unwrap();
+    assert_eq!(
+        market_start.asks.len(),
+        market_end.asks.len(),
+        "Order should have failed silently"
+    );
+
```

Comment:
here, let's send another order with fail_silently set to false, and check that tx fails